### DISCUSSION TITLE
Deindex entries if user is suspended or deleted

### DISCUSSION THREAD

**Discussion by zizther** [studioespresso/craft-scout#277] [Q&A] [answered]
Is there a way to or support deindexing of entries if a user is suspended or when a user is deleted (and entries are not passed to another user)?

My scenario is that a user is suspended because the violated the community rules, if a user is suspended their content is also not visible, it would also be good to deindex their entries so they are not discoverable.
We would want the entries to be reindexed if the user is then set back to active/is unsuspended.

The same goes for deindexing if a user is deleted.

Is there a way to support this in Scout or would it have to be a custom setup in a custom module?

**Comment by zizther** [comment]
@janhenckens Any thoughts on this or the possibility to use a method from the plugin via a custom module to update/remove/add an item to an index upon certain events where an element isn't necessarily being saved

**Comment by janhenckens** [comment, marked as answer]
`craft\services\Users` has events that run on suspend and unsuspend, both before and after so those should be what you need for this. 

To deindex an item you can call the `unsearchable()` behaviour on it, and to reindex them again if the account is restored can use `searchable()`.

Code-wise that could look like this:
````php
        Event::on(
            Users::class,
            Users::EVENT_AFTER_SUSPEND_USER,
            function (UserEvent $event) {
                $userContent = Entry::findAll(['relatedTo' => $event->user]);
                foreach($userContent as $entry){
                    $entry->unsearchable();
                }
            }
        );
````

**Reply by zizther** [reply]
Thanks, that is great. Do you know of it works the same for User Elements too?

Does Scout work off Craft making entries searchable or unsearchable?

**Reply by janhenckens** [reply]
You can make call `searchable` on a user but that will only work if the user is included in a query in the index settings - Doing `$user->unsearchable` won't automatically remove their content though, you'll have to have the relation yourself.

**Reply by zizther** [reply]
Great. In my case I have 2 indexes, one for entries in a section and another for users themselves.

I see, searchable and unsearchable are Scout functionality.

**Reply by janhenckens** [reply]
Yes, Scout adds those to any element (entry, user, category, etc).
In your case you could do this: 

```php
 Event::on(
            Users::class,
            Users::EVENT_AFTER_SUSPEND_USER,
            function (UserEvent $event) {
                // Remove user from Algolia
                $event->user->unsearchable();

                // Remove the user's content from Algolia
                $userContent = Entry::findAll(['relatedTo' => $event->user]);
                foreach($userContent as $entry){
                    $entry->unsearchable();
                }
            }
        );
```

**Reply by janhenckens** [reply]
@zizther could you mark this as the correct answer? Then I'll reopen it so it stays visible for other users. Thanks!

**Reply by zizther** [reply]
No probs, done so
